domain/controllerupgrader/state/types.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+// controllerNodeAgentVersion represents the agent version running for each
+// controller node in the cluster.
+type controllerNodeAgentVersion struct {
+	ControllerID string `db:"controller_id"`
+	Version      string `db:"version"`
+}
+
+// agentVersionTarget represents the target agent version column from the
+// agent_version table.
+type agentVersionTarget struct {
+	TargetVersion string `db:"target_version"`
+}
+
+// controllerTargetVersion represents the current target version set for the
+// controller.
+type controllerTargetVersion struct {
+	TargetVersion string `db:"target_version"`
+}
+
+// isControllerModel represents the is_controller_model column value from  the
+// model table in the controller's model database.
+type isControllerModel struct {
+	Is bool `db:"is_controller_model"`
```

Comment:
nit/suggestion (non-blocking): Would it be clearer if we name this field as IsControllerModel to be consistent with the db tag? Perhaps if we want to differentiate the struct and field we could set the struct as isModelControllerModel or isModelControllerScoped